behaviors/TranslatableCmsObject.php

```diff
@@ -43,6 +45,20 @@ public function __construct($model)
         $this->model->bindEvent('cmsObject.getTwigCacheKey', function($key) {
             return $this->overrideTwigCacheKey($key);
         });
+
+        $this->model->bindEvent('model.afterDelete', function() use ($model) {
+            $contentPath = sprintf('%s/%s/%s', themes_path(), $model->theme->getDirName(), $model->getObjectTypeDirName());
+
+            foreach (Locale::listEnabled() as $locale => $label) {
+                if ($locale == $this->translatableDefault) {
+                    continue;
+                }
+                $localePath = sprintf('%s-%s/%s', $contentPath, $locale, $model->fileName);
+                if (File::exists($localePath)) {
+                    File::delete($localePath);
+                }
```

Comment:
Don't use the file facade for this, get the relevant Halcyon object directly and call delete on that; that will ensure support for the upcoming DB layer. Also add a one line comment describing the purpose of this code.

Reply:
This should be the right way

Reply:
Did you test it?

Reply:
Why is it required to set the filename if you're calling `getCmsObjectForLocale` already?

Reply:
Yes, tested it and it works fine.

As for setting the filename, I replicated code from another method ( syncTranslatableFileNames ) which does the same thing... but you are right, this is not necessary.